src/tools/pbs_probe.c

```diff
@@ -2257,29 +2257,25 @@ get_realpath_values(struct infrastruct *pinf)
 	 * First try and resolve to a real path the MPUG path
 	 * data belonging to *pinf's "pri" member
 	 */
-
 	for (i=0; i<PBS_last; ++i) {
 
 		good_prime[i] = 0;
 		if (pinf->pri.pbs_mpug[i].path) {
-			if (realpath(pinf->pri.pbs_mpug[i].path, &real[0])) {
-				pinf->pri.pbs_mpug[i].realpath = strdup(&real[0]);
+			if ((real = realpath(pinf->pri.pbs_mpug[i].path, NULL)) != NULL) {
```

Comment:
Here you are allocating memory for real in a for() loop without freeing it.  Will it ever allocate twice and leak?  Also, I don't see a free() further down except in the middle of a for() loop.  Is it possible for this function to exit and leak it?